reactor-core/src/main/java/reactor/core/scheduler/BoundedElasticScheduler.java

```diff
@@ -160,61 +139,124 @@ BoundedScheduledExecutorService createBoundedExecutorService() {
 
 	@Override
 	public boolean isDisposed() {
-		return BOUNDED_SERVICES.get(this) == SHUTDOWN;
+		return state.currentResource == SHUTDOWN;
 	}
 
 	@Override
 	public void start() {
-		for (;;) {
-			BoundedServices services = BOUNDED_SERVICES.get(this);
-			if (services != SHUTDOWN) {
+		SchedulerState<BoundedServices> a = this.state;
```

Comment:
should not we keep for ;; loop here?

Reply:
I think it's not needed, can you elaborate?

Reply:
yeah. I changed my mind